client/src/components/attendance/AttendanceForm.js

```diff
@@ -145,27 +146,29 @@ export default function AttendanceForm(props) {
         setCurrentEvent(foundEvent)
       } else {
         // Get the corresponding event and set it to current event
-        fetch(`/api/events?eventId=${props.eventId}`, { method: 'GET' }).then(async (res) => {
+        fetch(`/api/events?eventId=${props.eventId}`, {method: 'GET'}).then(async (res) => {
           const data = await res.json()
           setCurrentEvent(data)
         })
       }
 
       // Get the corresponding attendance data and set
-      fetch(`/api/attendance?eventId=${props.eventId}`, { method: 'GET' }).then(async (res) => {
+      fetch(`/api/attendance?eventId=${props.eventId}`, {method: 'GET'}).then(async (res) => {
         const data = await res.json()
         setAttendanceData(data)
       })
     }
   }, [])
 
   const handleCurrentEventChange = (eventId) => {
-    setAttendanceData([])
-    setCurrentEvent(recentEvents.find(event => event._id === eventId))
-    fetch(`/api/attendance?eventId=${eventId}`, { method: 'GET' }).then(async (res) => {
-      const data = await res.json()
-      setAttendanceData(data)
-    })
+    if (recentEvents.find(event => event._id === eventId) !== undefined) {
+      setAttendanceData([])
+      setCurrentEvent(recentEvents.find(event => event._id === eventId))
```

Comment:
Bring `recentEvents.find(event => event._id === eventId)` out of the parameters as a separate const and then call it to avoid calling `...find()` function twice as it's more intensive.